sql/select.go

```diff
@@ -112,6 +121,7 @@ func (p *planner) expandSubqueries(stmt parser.Statement) error {
 func (p *planner) selectIndex(s *scanNode) (planNode, error) {
 	if s.desc == nil || s.filter == nil {
 		// No table or where-clause.
+		s.initOrdering()
```

Comment:
but why not do the `initOrdering` before the `selectIndex`? i might be missing something.